nni/algorithms/compression/v2/pytorch/utils/evaluator.py

```diff
@@ -0,0 +1,581 @@
+        else:
+            nni_metric = None
+        return nni_metric, original_results
+
+    def get_dummy_input(self) -> Any:
+        if self._dummy_input:
+            return self._dummy_input
+        return next(iter(self.data_module.train_dataloader()))
+
+
+_CRITERION = Callable[[Any, Any], Any]
+_EVALUATING_FUNC = Union[Callable[[Module], float], Callable[[Module], Tuple[float, Any]]]
+_TRAINING_FUNC = Callable[[Module, Union[Optimizer, List[Optimizer]], _CRITERION, Union[None, _LRScheduler, List[_LRScheduler]], Optional[int], Optional[int]], None]
+
+
+class TorchEvaluator(Evaluator):
+    """
+    Parameters
+    ----------
+    training_func
+        The training function is used to train the model, note that this a entire optimization training loop.
+        It should have three required parameters [model, optimizers, criterion] and three optional parameters [schedulers, max_steps, max_epochs].
+        ``optimizers`` can be an instance of ``torch.optim.Optimizer`` or a list of ``torch.optim.Optimizer``, it belongs to the ``optimizers`` pass to ``TorchEvaluator``.
+        ``criterion`` and ``schedulers`` are also belonging to the ``criterion`` and ``schedulers`` pass to ``TorchEvaluator``.
+        ``max_steps`` and ``max_epochs`` are used to control the training duration.
+
+        Example::
+
+            def training_func(model: Module, optimizer: Optimizer, criterion: Callable, scheduler: _LRScheduler,
+                              max_steps: int | None = None, max_epochs: int | None = None, *args, **kwargs):
+                model.train()
+
+                # prepare data
+                data_dir = Path(__file__).parent / 'data'
+                MNIST(data_dir, train=True, download=True)
+                transform = transforms.Compose([transforms.ToTensor(), transforms.Normalize((0.1307,), (0.3081,))])
+                mnist_train = MNIST(data_dir, train=True, transform=transform)
+                train_dataloader = DataLoader(mnist_train, batch_size=32)
+
+                max_epochs = max_epochs if max_epochs else 3
+                max_steps = max_steps if max_steps else 6000
+                current_steps = 0
+
+                # training
+                for _ in range(max_epochs):
+                    for x, y in train_dataloader:
+                        optimizer.zero_grad()
+                        x, y = x.to(device), y.to(device)
+                        logits = model(x)
+                        loss: torch.Tensor = criterion(logits, y)
+                        loss.backward()
+                        optimizer.step()
+                        current_steps += 1
+                        if max_steps and current_steps == max_steps:
+                            return
+                    scheduler.step()
+
+    optimziers
+        The traced optimizer instance which the optimizer class is wrapped by nni.trace.
+        E.g. ``traced_optimizer = nni.trace(torch.nn.Adam)(model.parameters())``.
```

Comment:
This will all appear in one line in the docs. Suggest numbering them with 1,2,3,4...

Reply:
how to use 1,2,3?

Reply:
Like in rst.